lib/primitives/base-element.js

```diff
@@ -35,6 +35,104 @@ module.exports = function(registry) {
       return this.content;
     },
 
+    /// Finds the given elements in the element tree
+    /// Returns ArrayElement of elements or null
+    /// findRecursive: function(names..., parents)
+    findRecursive: function() {
+      var elementNames = [].concat.apply([], arguments);
+      var ArrayElement = registry.getElementClass('array');
+      var array = new ArrayElement();
+      var parents;
+
+      if (elementNames[elementNames.length - 1].element === 'array') {
+        parents = elementNames.pop().clone();
```

Comment:
Otherwise, when we add our self (`this` element) a few lines down then we are mutating the given inherited parents array element which in turn will cause the current element to show up on other parent trees that are attach to other elements.

It was actually discussed about making getters/setters immutable too which could allow us to handle this differently in the future -> https://github.com/refractproject/minim/issues/13